Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -202,7 +201,11 @@ public NodejsProjectNode(NodejsProjectPackage package)
 
         public bool IsTypeScriptProject {
             get {
```

Comment:
I believe that this handled the main race condition. There are still a few potential problems, but these are much more corner cases.

The latest iteration check a project property to tell if we are in a typescript project or not. That handles the main race condition we were running into, since that property should be accurate, even before the project has fully loaded.

One edge case is when you have a JavaScript project with a typescript file in it. In that case, we mark the project as a typescript project. But before this takes place, we may try to acquire typings. I don't see any easy way around this case.